### ISSUE TITLE
ssl_verify_hostname sync with ruby-kafka/delivery_boy

### ISSUE THREAD

**Issue by chaadow** [karafka/waterdrop#103]
Hello,
(Follow up from [this thread](https://github.com/karafka/karafka/issues/506)

Here are the specs i'm using in my app:  `karafka 1.2.13, ruby-kafka 0.7.9, rails 5.2.3, ruby 2.6.3`

Karafka depends on waterdrop which depends on delivery_boy.
There is a new option added quite recently in ruby-kafka: `ssl_verify_hostname` which defaults to true ( both consumers and producers). On the consumer side, everything works good, karafka is in sync with ruby-kafka. And this can be disabled by changing the config in `karafka.rb`

I have a similar error message as the one described in this [issue](https://github.com/zendesk/ruby-kafka/issues/734).
`OpenSSL::SSL::SSLError: SSL_connect returned=1 errno=0 state=error: certificate verify failed (unspecified certificate verification error)`

Basically the idea is to sync waterdrop the same way karafka was synced by changing the "dry config" file and adding the same setting (`ssl_verify_hostname`). so that when the async producer is created, it can pass a different value other than the default.

Now delivery_boy has this "fix" but only on master (not released yet :/).

So to make this work with karafka,  all I had to do was fork delivery_boy, change the default from true to false, until delivery_boy releases the patch, and waterdrop gem bumps delivery boy. and karafka gem bumps waterdrop lol.

**Comment by chaadow** [comment]
@mensfeld I just remembered that I read tha you're transitioning from ruby-kafka(delivery boy/racecar) to rdkafka, this will not be an issue in the future I guess.

I asked on the ruby-kafka slack to bump delivery boy version. And then I'll happily submit a PR to this repo 👌

**Comment by mensfeld** [comment]
Hey. Please use the fork for now. The switch to rdkafka is already on my agenda as I need that for high-throughput work. The development will start next wekk :)

I will close that, as I don't plan to do any custom work with racecar anymore.
